src/ucs/debug/log.c

```diff
@@ -94,34 +115,127 @@ void ucs_log_flush()
 size_t ucs_log_get_buffer_size()
 {
     return ucs_config_memunits_get(ucs_global_opts.log_buffer_size,
-                                   256, 2048);
+                                   256, UCS_ALLOCA_MAX_SIZE);
+}
+
+static void ucs_log_get_file_name(char *log_file_name, size_t max, unsigned idx)
+{
+    ucs_assert(idx <= ucs_global_opts.log_file_backup_count);
+
+    if (idx == 0) {
+        ucs_strncpy_zero(log_file_name, ucs_log_file_names[0], max);
+        return;
+    }
+
+    ucs_snprintf_zero(log_file_name, max, "%s.%u",
+                      ucs_log_file_names[idx], idx);
+}
+
+static void ucs_log_handle_file_max_size(int log_entry_len)
+{
+    const char *next_token;
+    char old_log_file_name[UCS_LOG_MAX_FILE_NAME];
+    char new_log_file_name[UCS_LOG_MAX_FILE_NAME];
+    unsigned idx, start_idx;
+    int ret;
+
+    /* check if it is necessary to find a new storage for logs */
+    if ((log_entry_len + ftell(ucs_log_file)) >=
+        ucs_global_opts.log_file_max_size) {
+        fclose(ucs_log_file);
```

Comment:
pls put he code of 'if' to separate function

Reply:
done